ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/MemcacheClient.java

```diff
@@ -31,15 +38,38 @@
  */
 public class MemcacheClient<K, V> implements TypedCache<K, V> {
   private static final long MAX_EXPIRATION_SEC = 60 * 60 * 24 * 30;
+  private static final String COMPONENT = MemcacheClient.class.getSimpleName();
 
   private final MemcachedClientIF spyClient;
   private final CacheKeyTranslator<K> keyTranslator;
   private final int expirationSpyUnits;
 
-  public MemcacheClient(final MemcachedClientIF spyClient, final CacheKeyTranslator<K> keyTranslator, final long expiration, final TimeUnit timeUnit) {
+  private MetricFactory metricFactory = null;
```

Comment:
reject: I want to retain the old behavior in I don't get a metricFactory and thus don't report metrics. I use metricFactory as the generic flag (rather than a fine-grained checking for the existence (!= null) of each individual metric which would make the code more complicated.